packages/verified-fetch/src/verified-fetch.ts

```diff
@@ -275,17 +277,20 @@ export class VerifiedFetch {
     let terminalElement: UnixFSEntry | undefined
     let ipfsRoots: CID[] | undefined
     let redirected = false
+    const byteRangeContext = new ByteRangeContext(this.helia.logger, options?.headers)
 
     try {
       const pathDetails = await walkPath(this.helia.blockstore, `${cid.toString()}/${path}`, options)
       ipfsRoots = pathDetails.ipfsRoots
       terminalElement = pathDetails.terminalElement
     } catch (err) {
-      this.log.error('Error walking path %s', path, err)
+      this.log.error('error walking path %s', path, err)
+
+      return badGatewayResponse('Error walking path')
     }
 
     let resolvedCID = terminalElement?.cid ?? cid
-    let stat: UnixFSStats
+    let stat: UnixFSStats | undefined
```

Comment:
```suggestion
    let stat: UnixFSStats
```

Can probably remove this again since we only call stat once. If it requires no changes ill impl, otherwise ill leave it